openapi3.yaml

```diff
@@ -1,64 +1,101 @@
   version: 1.0.0
   license:
     name: MIT
     url: https://opensource.org/licenses/MIT
 
 paths:
-  /resourceName:
+  /{z}/{x}/{y}.png:
     get:
-      operationId: getResourceName
+      operationId: getTiles
       tags:
-        - resourceName
-      summary: gets the resource
+        - rasterize
+      summary: get raster tiles by z, x, y
+      parameters:
+        - in: path
+          name: 'z'
+          required: true
+          schema:
+            allOf:
+              - $ref: '#/components/schemas/tilePosition'
+              - maximum: 22
+          description: zoom level
+        - in: path
+          name: 'x'
+          required: true
+          schema:
+            $ref: '#/components/schemas/tilePosition'
+          description: x axis tile
+        - in: path
+          name: 'y'
+          required: true
+          schema:
+            $ref: '#/components/schemas/tilePosition'
+          description: y axis tile
       responses:
         200:
-          description: OK
+          description: the requested rasterized image
           content:
-            application/json:
+            image/png:
               schema:
-                $ref: '#/components/schemas/resource'
-    post:
-      operationId: createResource
-      tags:
-        - resourceName
-      summary: creates a new record of type resource
-      responses:
-        201:
-          description: created
-          content:
-            application/json:
-              schema:
-                $ref: '#/components/schemas/resource'
+                type: string
+                format: binary
         400:
-          description: Bad Request
-          content:
-            application/json:
-              schema:
-                $ref: '#/components/schemas/error'
+          $ref: '#/components/responses/BadRequest'
+        404:
+          $ref: '#/components/responses/NotFound'
+        '5XX':
+          $ref: '#/components/responses/UnexpectedError'
 components:
+  responses:
+    BadRequest:
+      description: Bad Request
+      content:
+        application/json:
+          schema:
+            allOf:
+              - $ref: '#/components/schemas/error'
+              - properties:
+                  statusCode:
+                    enum:
+                      - 400
+    NotFound:
+      description: Not Found
+      content:
+        application/json:
+          schema:
+            allOf:
+              - $ref: '#/components/schemas/error'
+              - properties:
+                  statusCode:
+                    enum:
+                      - 404
+    UnexpectedError:
+      description: Unexpected Error
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/error'
   schemas:
+    tilePosition:
+      type: number
+      format: int32
+      default: 0
+      minimum: 0
     error:
       type: object
       required:
         - message
+        - error
```

Comment:
what is the difference between `error` and message `fields`?

Reply:
[see this](https://www.fastify.io/docs/latest/Reply/#errors), for some reason the field `code` is not being returned. not that we want it.
so there are three fields. `error` is the http status code in words while `message` is a custom message
```json
{
  "statusCode": 500,
  "error": "Internal Server Error",
  "message": "blabla"
}`